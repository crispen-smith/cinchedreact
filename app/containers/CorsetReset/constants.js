import keymirror from 'keymirror';

export const CORSET_RESET_ACTION_TYPES = keymirror({
  default: null,
  reset: null,
});
