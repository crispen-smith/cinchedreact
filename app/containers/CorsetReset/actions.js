/*
 *
 * CorsetReset actions
 * 
 * Note that CorsetReset is intended only for Dev purposes}
 * 
 *
 */

import { CORSET_RESET_ACTION_TYPES } from './constants';

export function defaultAction() {
  return {
    type: CORSET_RESET_ACTION_TYPES.default,
  };
}

export function resetAction() {
  return {
    type: CORSET_RESET_ACTION_TYPES.reset,
  };
}
