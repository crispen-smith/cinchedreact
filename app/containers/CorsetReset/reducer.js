/*
 *
 * CorsetReset reducers
 * 
 * Reset the corsets gallery for dev purposes to allow for a clean slate
 * This may or may not need to be hooked into the Stitch app but for now we just need
 * to fight persistance
 *
 */

import { fromJS } from 'immutable';
import { CORSET_RESET_ACTION_TYPES } from './constants';

export const initialState = fromJS({});

function corsetResetReducer(state = initialState, action) {
  switch (action.type) {
    case CORSET_RESET_ACTION_TYPES.reset:
      return resetCorsets(state);
    case CORSET_RESET_ACTION_TYPES.default:
      return state;
    default:
      return state;
  }
}

const resetCorsets = state => {
  const newState = state.toJS();
  if (newState.corsets) newState.corsets = [];
  return fromJS(newState);
};

export default corsetResetReducer;
