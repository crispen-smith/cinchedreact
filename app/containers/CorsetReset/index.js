/*
/
/ Corset Reset - when the gallery is all F'ed up just reset it.
/ Break *ONLY* in case of emergency
/
*/

// Import a whole bunch of stuff we don't need,
// TODO: TRIM THIS LIST!

import React from 'react';
import PropTypes from 'prop-types';

import { connect } from 'react-redux';

import { compose } from 'redux';
import { createStructuredSelector } from 'reselect';

import injectReducer from 'utils/injectReducer';

import { makeSelectLoggedIn } from '../BrandedHeader/selectors';
import { makeSelectCorsetGallery } from '../CorsetGallery/selectors';

import reducer from './reducer';

import * as actions from './actions';

/* eslint-disable react/prefer-stateless-function */
class CorsetReset extends React.Component {
  constructor(props) {
    super(props);
    if (!this.props.loggedIn) return;
    const { dispatch } = this.props;
    const resetAction = actions.resetAction();
    dispatch(resetAction);
  }

  render() {
    return <div>CORSET RESET</div>;
  }
}

CorsetReset.propTypes = {
  loggedIn: PropTypes.bool.isRequired,
  dispatch: PropTypes.func.isRequired,
};

const mapStateToProps = createStructuredSelector({
  corsetGallery: makeSelectCorsetGallery(),
  loggedIn: makeSelectLoggedIn(),
});

function mapDispatchToProps(dispatch) {
  return {
    dispatch,
  };
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

const withReducer = injectReducer({ key: 'corsetGallery', reducer });

export default compose(
  withReducer,
  withConnect,
)(CorsetReset);
