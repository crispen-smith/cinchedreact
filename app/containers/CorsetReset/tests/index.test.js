import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import CorsetReset from '../index';
import configureStore from '../../../configureStore';

describe('<CorsetReset />', () => {
  it('Renders a warning message when loggedIn = False', () => {
    const brand = { loggedIn: false };
    const match = { params: {} };
    const store = configureStore({}, browserHistory, brand);
    const reset = mount(
      <Provider store={store}>
        <CorsetReset match={match} />
      </Provider>,
    );

    expect(reset).toMatchSnapshot();
  });
  it('Renders the Reset Message when loggedIn = True', () => {
    const brand = { loggedIn: true };
    const match = { params: {} };
    const store = configureStore({}, browserHistory, brand);
    const reset = mount(
      <Provider store={store}>
        <CorsetReset match={match} />
      </Provider>,
    );

    expect(reset).toMatchSnapshot();
  });
});
