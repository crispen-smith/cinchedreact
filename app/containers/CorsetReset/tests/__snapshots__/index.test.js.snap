// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CorsetReset /> Renders a warning message when loggedIn = False 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "injectedReducers": Object {
        "corsetGallery": [Function],
      },
      "injectedSagas": Object {},
      "replaceReducer": [Function],
      "runSaga": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <withReducer(Connect(CorsetReset))
    match={
      Object {
        "params": Object {},
      }
    }
  >
    <Connect(CorsetReset)
      match={
        Object {
          "params": Object {},
        }
      }
    >
      <CorsetReset
        corsetGallery={
          Object {
            "corsets": Array [],
          }
        }
        dispatch={[Function]}
        loggedIn={true}
        match={
          Object {
            "params": Object {},
          }
        }
      >
        <h1>
          CORSET RESET
        </h1>
        <span>
          Gallery has been Reset
        </span>
      </CorsetReset>
    </Connect(CorsetReset)>
  </withReducer(Connect(CorsetReset))>
</Provider>
`;

exports[`<CorsetReset /> Renders the Reset Message when loggedIn = True 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "injectedReducers": Object {
        "corsetGallery": [Function],
      },
      "injectedSagas": Object {},
      "replaceReducer": [Function],
      "runSaga": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <withReducer(Connect(CorsetReset))
    match={
      Object {
        "params": Object {},
      }
    }
  >
    <Connect(CorsetReset)
      match={
        Object {
          "params": Object {},
        }
      }
    >
      <CorsetReset
        corsetGallery={
          Object {
            "corsets": Array [],
          }
        }
        dispatch={[Function]}
        loggedIn={true}
        match={
          Object {
            "params": Object {},
          }
        }
      >
        <h1>
          CORSET RESET
        </h1>
        <span>
          Gallery has been Reset
        </span>
      </CorsetReset>
    </Connect(CorsetReset)>
  </withReducer(Connect(CorsetReset))>
</Provider>
`;
